app/src/main/res/layout/fragment_garden.xml

```diff
@@ -36,20 +36,38 @@
                 android:layout_width="match_parent"
                 android:layout_height="match_parent"
                 android:clipToPadding="false"
-                android:paddingLeft="@dimen/margin_normal"
-                android:paddingRight="@dimen/margin_normal"
+                android:layout_marginStart="@dimen/card_side_margin"
+                android:layout_marginEnd="@dimen/card_side_margin"
+                android:layout_marginTop="@dimen/margin_normal"
                 app:isGone="@{!hasPlantings}"
-                app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                app:layoutManager="androidx.recyclerview.widget.StaggeredGridLayoutManager"
+                app:spanCount="@integer/grid_columns"
                 tools:listitem="@layout/list_item_garden_planting"/>
 
-        <TextView
-                android:id="@+id/empty_garden"
+        <LinearLayout
                 android:layout_width="match_parent"
                 android:layout_height="match_parent"
                 android:gravity="center"
-                android:text="@string/garden_empty"
-                android:textSize="24sp"
-                app:isGone="@{hasPlantings}"/>
+                android:orientation="vertical"
+                app:isGone="@{hasPlantings}">
+
+            <TextView
+                    android:id="@+id/empty_garden"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="@string/garden_empty"
+                    android:textAppearance="?attr/textAppearanceHeadline5" />
+
+            <com.google.android.material.button.MaterialButton
+                android:id="@+id/add_plant"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:backgroundTint="?attr/colorOnPrimary"
```

Comment:
The Attribute 'backgroundTint' is only used in API level 21 and higher (current min is 19). For solve that exist different ways, avoiding to break design for lower API versions:

- Define a DrawableBackground with rounded borders and assign as a background
- Create an button style with the property _backgroundTint_ and adding at button style
- Using background set color filter

Reply:
This is back-ported in appcompat but you need to change the ns.

```suggestion
                app:backgroundTint="?attr/colorOnPrimary"
```

Reply:
Thanks for pointing this out @Vmadalin! Applying Nick's suggested fix.